server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java

```diff
@@ -434,8 +435,8 @@ private static long getExpectedTotalHitCountWithShortcut(IndexReader reader, int
      */
     public void testTerminateAfterSize0NoHitCountShortcut() throws Exception {
         indexDocs();
-        BooleanQuery bq = new BooleanQuery.Builder().add(new TermQuery(new Term("foo", "bar")), Occur.SHOULD)
-            .add(new TermQuery(new Term("foo", "baz")), Occur.SHOULD)
+        BooleanQuery bq = new BooleanQuery.Builder().add(new NonCountingTermQuery(new Term("foo", "bar")), Occur.SHOULD)
+            .add(new NonCountingTermQuery(new Term("foo", "baz")), Occur.SHOULD)
```

Comment:
I guess we can simplify this to not have a bool query? Can you do the same for the other one that ends with NoHitCountShortcut please?

Reply:
Sure I can, I was under the impression that the bool query is needed here for other reasons.